### PR TITLE
kubeadm: mark the phase 'experimental-cert-rotation' as deprecated

### PR DESCRIPTION
**Pull request by neolit123**
#### What type of PR is this?

/kind cleanup deprecation

#### What this PR does / why we need it:

- Mark the phase 'init kubelet-finilize experimental-cert-rotation' as deprecated and print a warning if its used directly. It should be removed in 1.32.
- Add a replacement phase 'enable-client-cert-rotation'.

(!!) We should stop using the 'experimental' and 'alpha' words in phases, clI flags and k8s object names as this complicates the graduation and users will use the feature in production regardless.

Having a 'DEPRECATED' or 'EXPERIMENTAL' text in descriptions is sufficient to indicate a status of a feature that should not be used in production.

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/kubeadm/issues/3046

#### Special notes for your reviewer:
NONE

#### Does this PR introduce a user-facing change?

```release-note
kubeadm: mark the sub-phase of 'init kubelet-finilize' called 'experimental-cert-rotation' as deprecated and print a warning if it is used directly; it will be removed in a future release. Add a replacement sub-phase 'enable-client-cert-rotation'.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by neolit123** [comment]
searching on github i found a couple of shell scripts that use the deprecated phase directly.
we don't mention it in authored docs at k8s.io. it's only listed in generated reference docs.

**Comment by pacoxu** [comment]
> searching on github i found a couple of shell scripts that use the deprecated phase directly.

Out of `kubernetes` community? 

> the plan is to remove it in 1.32.

That's OK for deprecating an experimental phase. Or a later release. Either is OK to me.

LGTM

leave to @SataQiu to take a look.

**Comment by neolit123** [comment]
> Out of kubernetes community?

not from k / k-sigs orgs.

> That's OK for deprecating an experimental phase. Or a later release. Either is OK to me.

as per the rules we can remove it without deprecation. but if something says alpha or expirimental for a long time and is used in user scripts and production, then it's not alpha / experimental anymore. a great example is kube-proxy's v1alpha1 config.

**Comment by pacoxu** [comment]
> leave to @SataQiu to take a look.

/lgtm
/hold
double checked

File: cmd/kubeadm/app/cmd/phases/init/kubeletfinalize.go [hunk 1]
```diff
@@ -57,19 +60,48 @@ func NewKubeletFinalizePhase() workflow.Phase {
...
+// runKubeletFinalizeEnableClientCertRotationWrapped wraps runKubeletFinalizeEnableClientCertRotation
+// and prints a deprecation message when the phase is executed directly. If 'all' is used this
+// function should just return nil because 'enable-client-cert-rotation' sets 'enableClientCertRotationRun'.
+// TODO: remove in 1.32.
+// https://github.com/kubernetes/kubeadm/issues/3046
+func runKubeletFinalizeEnableClientCertRotationWrapped(c workflow.RunData) error {
+	if enableClientCertRotationRun {
+		return nil
+	}
+	klog.Warning("The phase 'experimental-cert-rotation' is deprecate and will be removed in a future release. " +
```
Comment (SataQiu) [inline cmd/kubeadm/app/cmd/phases/init/kubeletfinalize.go:90]:
```suggestion
	klog.Warning("The phase 'experimental-cert-rotation' is deprecated and will be removed in a future release. " +
```

**Review by SataQiu — COMMENTED** [review]
SGTM

File: cmd/kubeadm/app/cmd/phases/init/kubeletfinalize.go (hunk 1 above)
Comment (neolit123) [reply cmd/kubeadm/app/cmd/phases/init/kubeletfinalize.go]: updated.

**Review by SataQiu — COMMENTED** [review]
/lgtm

**Pull request merged**